src/EditorFeatures/Core/Implementation/Classification/SyntacticClassificationTaggerProvider.TagComputer.cs

```diff
@@ -135,43 +149,72 @@ private void ConnectToWorkspace(Workspace workspace)
                 ResetLastParsedDocument();
 
                 _workspace = workspace;
-                _workspace.WorkspaceChanged += this.OnWorkspaceChanged;
-                _workspace.DocumentOpened += this.OnDocumentOpened;
-                _workspace.DocumentActiveContextChanged += this.OnDocumentActiveContextChanged;
 
-                var textContainer = _subjectBuffer.AsTextContainer();
+                if (_workspace != null)
+                {
+                    _isClassificationOnlyWorkspace = false;
+                    _workspace.WorkspaceChanged += this.OnWorkspaceChanged;
+                    _workspace.DocumentOpened += this.OnDocumentOpened;
+                    _workspace.DocumentActiveContextChanged += this.OnDocumentActiveContextChanged;
+
+                    var textContainer = _subjectBuffer.AsTextContainer();
 
-                var documentId = _workspace.GetDocumentIdInCurrentContext(textContainer);
-                if (documentId != null)
+                    var documentId = _workspace.GetDocumentIdInCurrentContext(textContainer);
+                    if (documentId != null)
+                    {
+                        var document = workspace.CurrentSolution.GetDocument(documentId);
+                        if (document != null)
+                        {
+                            EnqueueParseSnapshotTask(document);
+                            return;
+                        }
+                    }
+                }
+                else
                 {
-                    var document = workspace.CurrentSolution.GetDocument(documentId);
-                    if (document != null)
+                    // This buffer isn't being tracked by any workspace. In Visual Studio, this
+                    // can happen if it is not in the Running Document Table. We'll try to do
```

Comment:
I often find myself looking at some workaround code wondering "What's the *real* reason they did this, so that I can hit this codepath in the product" -- Do you have thoughts on how to capture that intent without muddying the waters here?